### DISCUSSION TITLE
Clentaminator problem.

### DISCUSSION THREAD

**Discussion by Zetofu** [Pryaxis/TShock#3026] [General]
Hello, im having an issue where the clentaminator is really buggy like, whenever you spray at a corrupted block at a certain range it immediately converts to its corrupted state again rather than getting cleansed.

**Comment by punchready** [comment]
I'm unable to reproduce this on a fresh tShock install with everything left default while in the guest group (same with the Terraformer):
![image](https://github.com/Pryaxis/TShock/assets/22683812/d1547e04-d7b7-4e69-9752-991c4966f7ed)

As of the latest update to tile rect handling done by me, the Clentaminator also does not have a range check, so this is expected to work.

**Reply by Zetofu** [reply]
This is what happens when in default group.
https://github.com/Pryaxis/TShock/assets/160848497/305adb1b-94e2-4c32-86e2-274de77ce006

**Reply by punchready** [reply]
In a fresh install, the guest group has the least permissions, even less than the default group, which is why i chose it.
Can you set `DebugLogs` to `true` in `config.json`, reload the server (`/reload` or just restart it), then reproduce the issue and send the generated log file? Logs should be in something like `/tshock/logs` by default.

**Reply by Zetofu** [reply]
![image](https://github.com/Pryaxis/TShock/assets/160848497/f0090480-295b-4646-bbbe-dfbc98e56600)

**Reply by Zetofu** [reply]
[2024-03-01_16-16-27.log](https://github.com/Pryaxis/TShock/files/14458368/2024-03-01_16-16-27.log)
 
 
 heres the whole log

**Reply by Zetofu** [reply]
https://github.com/Pryaxis/TShock/assets/160848497/501ce0f6-23eb-4195-9bba-d09727436c20
[2024-03-01_16-21-07.log](https://github.com/Pryaxis/TShock/files/14458423/2024-03-01_16-21-07.log)

Oh and btw heres the second issue and the logs for it. and, its about dart traps.

**Comment by punchready** [comment]
It turns out that there have been no new releases after the fixes to this were merged, which means your installation doesn't contain them.
You can download a beta version from a CI run of your choice, for example https://github.com/Pryaxis/TShock/actions/runs/7869960417. Just scroll all the way down and download the right artifact for your system. Alternatively, you can follow the build instructions in the repository and build the latest nightly version yourself.
This should fix both of the issues you are experiencing.

**Reply by Zetofu** [reply]
aight thanks! it worked.
